src/wms/src/test/java/org/geoserver/wms/capabilities/GetCapabilitiesTransformerTest.java

```diff
@@ -415,6 +431,89 @@ private void checkVendorSpecificCapsProviders(GetCapabilitiesTransformer tr) thr
         assertEquals(1, list.getLength());
     }
 
+    /**
+     * Gets a capabilities document as an XML string Also, adds: <!DOCTYPE WMT_MS_Capabilities
+     * SYSTEM "WMS_MS_Capabilities.dtd"> at the top of the document.
+     */
+    String getCapabilitiesXML() throws Exception {
+        // WMS wms = new WMS(getGeoServer());
+        WMS wms = (WMS) applicationContext.getBean("wms");
+
+        GetCapabilitiesTransformer tr =
+                new GetCapabilitiesTransformer(wms, baseUrl, mapFormats, legendFormats, null);
+
+        Document dom = WMSTestSupport.transform(req, tr);
+        TransformerFactory ttf = TransformerFactory.newInstance();
+        Transformer trans = ttf.newTransformer();
+        StringWriter sw = new StringWriter();
+        trans.transform(new DOMSource(dom), new StreamResult(sw));
+
+        String xml = sw.toString();
+        xml =
+                xml.replace(
+                        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+                        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+                                + "<!DOCTYPE WMT_MS_Capabilities SYSTEM \"WMS_MS_Capabilities.dtd\">");
+
+        return xml;
+    }
+
+    @Test
+    public void testValidatesAgainstDTD() throws Exception {
+        // get a capabilities document
+        String getCapXML = getCapabilitiesXML();
+
+        // get the wms 1.1.1 DTD
+        URL dtdURL =
+                GetCapabilitiesTransformer.class.getResource(
+                        "/schemas/wms/1.1.1/wms_ms_capabilities.dtd");
+        String dtd = Resources.toString(dtdURL, StandardCharsets.UTF_8);
```

Comment:
I see this error in the logs:
2022-11-25T01:25:32.0940789Z 01:25:32,093 [ERROR] testValidatesAgainstDTD(org.geoserver.wms.capabilities.GetCapabilitiesTransformerTest)  Time elapsed: 0.836 s  <<< ERROR!
2022-11-25T01:25:32.0941352Z java.lang.NullPointerException
2022-11-25T01:25:32.0942044Z 	at org.geoserver.wms.capabilities.GetCapabilitiesTransformerTest.testValidatesAgainstDTD(GetCapabilitiesTransformerTest.java:470)

It works on Mac and Windows, but not on linux.

Reply:
I think this is the issue - I don't think the other issues you looked at are caused by this PR.  
Please see the linux output of other (recent) PR builds...

Reply:
Yes; windows and macOS are case in-sensitive. Linux is the sensitive one....